### PR TITLE
Fix some block entities not rendering in tent preview

### LINKED ISSUE #167: Some blocks don't render in the tent preview
java_crGyyw4uOg

### PR DESCRIPTION
**Pull request by jhbuchanan45**
Block ID required in the nbt. Our block entity toTag() implementation mistakenly did this when the ID should only be added when the createNbtWithId() method is called.

Partially closes #167 
I have also extensively looked into the grass blocks not rendering. It turns out any block rendered on the CUTOUT_MIPPED layer has this issue, and it has to do with our TentPreviewBufferBuilder vertex method multiplying the alpha value. I have *no* idea why this matters, someone more knowledgeable in shaders and minecraft rendering may know.

### CODE REVIEW

**Comment by Wyn-Price** [comment]
I noticed that normal CUTOUT *does* work, while CUTOUT_MIPPED does not. I plan to dive into the shader code after work today

**Review by Wyn-Price — APPROVED** [review]

File: src/main/java/com/terraformersmc/campanion/blockentity/TentPartBlockEntity.java [hunk 1]
```diff
@@ -34,7 +34,6 @@ public BlockPos getSize() {
 
 	@Override
 	public void toTag(NbtCompound tag) {
-		((InvokerBlockEntity) this).callWriteIdentifyingData(tag);
```
Comment (Wyn-Price) [inline src/main/java/com/terraformersmc/campanion/blockentity/TentPartBlockEntity.java:37]: Could you remove the mixin as I don't believe we need it any longer

**Review by Wyn-Price — CHANGES_REQUESTED** [review]

**Comment by Wyn-Price** [comment]
A PR fixing `cutout_mipped` can be found here #169, once you remove that mixin, I'll merge and push a change to curseforge. Thanks for the help!

**Comment by jhbuchanan45** [comment]
> A PR fixing `cutout_mipped` can be found here #169, once you remove that mixin, I'll merge and push a change to curseforge. Thanks for the help!

Great, this PR should be ready now

**Review by Wyn-Price — APPROVED** [review]

**Pull request merged**